geoips/create_plugin_registries.py

```diff
@@ -14,10 +14,128 @@
 import logging
 from geoips.commandline.log_setup import setup_logging
 import geoips.interfaces
+from geoips.errors import PluginRegistryError
 
 LOG = logging.getLogger(__name__)
 
 
+def remove_registries(plugin_packages):
+    """Remove all plugin registries if a PluginRegistryError is raised.
+
+    Parameters
+    ----------
+    plugin_packages: list EntryPoints
+        A list of EntryPoints pointing to each installed
+        GeoIPS package --> ie.
+        [EntryPoint(name='geoips', value='geoips', group='geoips.plugin_packages'), ...]
+    """
+    from os import remove
+
+    LOG.interactive(
+        "\n\n\n\nERROR: Removing registries due to duplicate plugins. You must fix the "
+        "error shown below before GeoIPS can operate correctly.\n\n\n"
+    )
+    for pkg in plugin_packages:
+        remove(str(resources.files(pkg.value) / "registered_plugins.yaml"))
+
+
+def registry_sanity_check(plugin_packages):
+    """Check that each plugin package registry has no duplicate lowest depth entries.
+
+    If it does, raise a PluginRegistryError for that specific package, then remove all
+    plugin registries from each package so the user must fix the error before
+    continuing.
+
+    Parameters
+    ----------
+    plugin_packages: list EntryPoints
+        A list of EntryPoints pointing to each installed
+        GeoIPS package --> ie.
+        [EntryPoint(name='geoips', value='geoips', group='geoips.plugin_packages'), ...]
+    """
+    for comp_idx, comp_pkg in enumerate(plugin_packages):
+        comp_registry = yaml.safe_load(
+            open(resources.files(comp_pkg.value) / "registered_plugins.yaml")
+        )
+        for pkg_idx, pkg in enumerate(plugin_packages):
+            if pkg_idx <= comp_idx:  # same package or already compared
+                continue
```

Comment:
Could you comment in the code a bit what all these loops are doing?  ie, comp_idx, pkg_idx, etc?

Reply:
Yup! I can add some more comments.

Reply:
Looks good to me, as long as @jsolbrig is good with how this is laid out.  It is a lot of looping, but it is only done once at install time so is probably fine?

Reply:
Yeah, I think this looks good for now. I expect a lot of this functionality will move into `PluginRegistry` when we do #389.